dot/peerset/peerstate.go

```diff
@@ -180,69 +181,99 @@ func (ps *PeersState) peers() []peer.ID {
 
 // sortedPeers returns the list of peers we are connected to of a specific set.
 func (ps *PeersState) sortedPeers(idx int) peer.IDSlice {
-	if len(ps.sets) < idx {
+	ps.RLock()
+	defer ps.RUnlock()
+
+	if len(ps.sets) == 0 || len(ps.sets) < idx {
 		logger.Debug("peer state doesn't have info for the provided index")
 		return nil
 	}
 
-	type kv struct {
-		peerID peer.ID
-		Node   *node
+	type connectedPeerReputation struct {
+		peerID     peer.ID
+		reputation Reputation
 	}
 
-	var ss []kv
-	for k, v := range ps.nodes {
-		state := v.state[idx]
+	connectedPeersReps := make([]connectedPeerReputation, 0)
```

Comment:
```suggestion
	connectedPeersReps := make([]connectedPeerReputation, 0, len(ps.nodes))
```

or just don't use a `make`.